### PR TITLE
FIX /repos API + modified_since wrong and inconsistent results

### PR DESCRIPTION
**Pull request by blayson**

### CODE REVIEW

File: webapp/repos.py [hunk 1]
```diff
@@ -57,6 +57,8 @@ def _filter_modified_since(self, repos_to_process, modified_since_dt):
         for label in repos_to_process:
             for repo_id in self.cache.repolabel2ids.get(label, []):
                 repo_detail = self.cache.repo_detail[repo_id]
+                if label in filtered_repos_to_process:
+                    continue
```
Comment (eherget) [inline webapp/repos.py]: I think the 'if label in filtered_repos_to_process:' could be moved before the repo_detail assignment statement above... if we're going to skip this label, might as well avoid doing anything else.  :)

Comment (blayson) [reply webapp/repos.py]: Sure :)

File: webapp/repos.py [hunk 2]
```diff
@@ -56,6 +56,8 @@ def _filter_modified_since(self, repos_to_process, modified_since_dt):
         filtered_repos_to_process = []
         for label in repos_to_process:
             for repo_id in self.cache.repolabel2ids.get(label, []):
+                if label in filtered_repos_to_process:
```
Comment (jdobes) [inline webapp/repos.py]: seems working but shouldn't this condition be rather in parent for cycle?

this way it's going to do many unnecessary loops

Comment (blayson) [reply webapp/repos.py]: Yes, done

**Pull request merged**